### PR TITLE
Replace Validations

### LINKED ISSUE #1474: Replace ember-validator with ember-cp-validations
https://github.com/offirgolan/ember-cp-validations is going to solve several problems we have with our existing validation library.  Mainly that we have had to disable several tests where it blows them up.

### PR DESCRIPTION
**Pull request by jrjohnson**
Working PR to replace our validation library with a much better one that doesn't break tests.
Fixes #1474

-  [x] app/components/course-header.js
-  [x] app/components/course-overview.js
-  [x] app/components/inplace-textarea.js
-  [x] app/components/new-course.js
-  [x] app/components/new-directory-user.js
-  [x] app/components/new-instructorgroup.js
-  [x] app/components/new-learnergroup-multiple.js
-  [x] app/components/new-learnergroup-single.js
-  [x] app/components/new-learnergroup.js
-  [x] app/components/new-learningmaterial.js
-  [x] app/components/new-myreport.js
-  [x] app/components/new-program.js
-  [x] app/components/new-program.js
-  [x] app/components/new-session.js
-  [x] app/components/new-user.js
-  [x] app/components/offering-editor.js
-  [x] app/components/offering-manager.js
-  [x] app/components/program-header.js
-  [x] app/components/program-overview.js
-  [x] app/components/school-list.js
-  [x] app/components/school-manager.js
-  [x] app/components/session-overview.js
-  [x] app/validators/local/alphanumeric.js
-  [x] app/validators/local/alphanumeric2.js
-  [x] app/validators/local/url.js

### CODE REVIEW

**Comment by jrjohnson** [comment]
Ok now waiting on tests - hopefully not too many failures...

**Comment by jrjohnson** [comment]
All that is left is to align the translations for the new error messages.  So close...

File: app/locales/es/translations.js [hunk 1]
```diff
@@ -483,36 +477,38 @@ export default {
...
+    'description': 'This field',
+    'inclusion': '{{description}} no está incluido en la lista',
+    'exclusion': '{{description}} es reservado',
+    'invalid': '{{description}} es inválido',
+    'confirmation': '{{description}} no coincide con {{on}}',
+    'accepted': '{{description}} debe ser aceptado',
+    'empty': '{{description}} no puede estar vacío',
+    'blank': '{{description}} no puede estar en blanco',
+    'present': '{{description}} debe estar en blanco',
+    'collection': '{{description}} must be a collection',
```
Comment (jrjohnson) [inline app/locales/es/translations.js]: Needs translation

File: app/locales/es/translations.js [hunk 2]
```diff
@@ -483,36 +477,38 @@ export default {
...
+    'singular': '{{description}} can not be a collection',
```
Comment (jrjohnson) [inline app/locales/es/translations.js]: Needs translation

File: app/locales/es/translations.js [hunk 3]
```diff
@@ -483,36 +477,38 @@ export default {
...
+    'tooLong': '{{description}} es demasiado largo (el máximo es {{max}} caracteres)',
+    'tooShort': '{{description}} es demasiado corto (el mínimo es {{min}} caracteres)',
+    'before': '{{description}} must be before {{before}}',
```
Comment (jrjohnson) [inline app/locales/es/translations.js]: Needs translation

File: app/locales/es/translations.js [hunk 4]
```diff
@@ -483,36 +477,38 @@ export default {
...
+    'after': '{{description}} must be after {{after}}',
```
Comment (jrjohnson) [inline app/locales/es/translations.js]: Needs translation

File: app/locales/es/translations.js [hunk 5]
```diff
@@ -483,36 +477,38 @@ export default {
...
+    'wrongDateFormat': '{{description}} must be in the format of {{format}}',
```
Comment (jrjohnson) [inline app/locales/es/translations.js]: Needs translation

File: app/locales/es/translations.js [hunk 6]
```diff
@@ -483,36 +477,38 @@ export default {
...
+    'greaterThan': '{{description}} debe ser mayor que {{gt}}',
+    'greaterThanOrEqualTo': '{{description}} debe ser mayor que o igual a {{gte}}',
+    'equalTo': '{{description}} debe ser igual a {{is}}',
+    'lessThan': '{{description}} debe ser menos que {{lt}}',
+    'lessThanOrEqualTo': '{{description}} debe ser menos que o igual a {{lte}}',
+    'otherThan': '{{description}} Debe ser otra cosa que {{value}}',
+    'odd': '{{description}} debe ser número impar',
+    'even': '{{description}} debe ser número par',
+    'positive': '{{description}} must be positive',
+    'date': '{{description}} must be a valid date',
```
Comment (jrjohnson) [inline app/locales/es/translations.js]: Needs translation

File: app/locales/es/translations.js [hunk 7]
```diff
@@ -483,36 +477,38 @@ export default {
...
-    'alphanumeric': "debe ser alfanumérico",
-    'alphanumeric2': "debe ser alfanumérico ('-' y ':' permitido)",
-    'url': 'debe ser una dirección URL válida',
-    'email': 'debe ser una dirección válida de correo electrónico',
-  },
-  'myreport': {
-    'errors': {
-      'titleTooLong':  "Título de Reporte es demasiado largo (el máximo es 240 caracteres)"
-    }
+    'description': 'This field',
```
Comment (saschaben) [inline app/locales/es/translations.js]: 'Este campo'

File: app/locales/es/translations.js (hunk 1 above)
Comment (saschaben) [inline app/locales/es/translations.js]: debe estar una collección

File: app/locales/es/translations.js (hunk 2 above)
Comment (saschaben) [inline app/locales/es/translations.js]: no pueded estar una collección

File: app/locales/es/translations.js (hunk 1 above)
Comment (thecoolestguy) [inline app/locales/es/translations.js]: 'collection': '{{description}} necesita estar una colección',

File: app/locales/es/translations.js (hunk 2 above)
Comment (thecoolestguy) [inline app/locales/es/translations.js]: 'singular': '{{description}} no se puede ser una colección',

File: app/locales/es/translations.js (hunk 3 above)
Comment (thecoolestguy) [inline app/locales/es/translations.js]: 'before': '{{description}} necesita estar antes de {{before}}',

File: app/locales/es/translations.js (hunk 4 above)
Comment (thecoolestguy) [inline app/locales/es/translations.js]: 'after': '{{description}} necesita estar después de {{after}}',

File: app/locales/es/translations.js (hunk 6 above)
Comment (thecoolestguy) [inline app/locales/es/translations.js]: 'date': '{{description}} necesita estar una fecha válida'

File: app/locales/es/translations.js (hunk 5 above)
Comment (thecoolestguy) [inline app/locales/es/translations.js]: 'wrongDateFormat': '{{description}} necesita ser en el formato de {{format}}',

File: app/locales/es/translations.js [hunk 8]
```diff
@@ -483,36 +477,38 @@ export default {
...
+    'equalTo': '{{description}} debe ser igual a {{is}}',
+    'lessThan': '{{description}} debe ser menos que {{lt}}',
+    'lessThanOrEqualTo': '{{description}} debe ser menos que o igual a {{lte}}',
+    'otherThan': '{{description}} Debe ser otra cosa que {{value}}',
+    'odd': '{{description}} debe ser número impar',
+    'even': '{{description}} debe ser número par',
+    'positive': '{{description}} must be positive',
+    'date': '{{description}} must be a valid date',
+    'email': '{{description}} debe ser una dirección válida de correo electrónico',
+    'phone': '{{description}} must be a valid phone number',
```
Comment (jrjohnson) [inline app/locales/es/translations.js]: @thecoolestguy Needs translation

**Pull request merged**